hooks/useClickOutside.ts

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useRef } from 'react';
+
+type useClickOutsideType = (_event: MouseEvent | TouchEvent) => void;
```

Comment:
```suggestion
type UseClickOutsideHandler = (_event: MouseEvent | TouchEvent) => void;
```

Reply:
This could be fine tuned by doing:

```suggestion
type UseClickOutsideHandler<T extends HTMLElement> = (_event: MouseEvent<T> | TouchEvent<T>) => void;
```